src/local_sensitivity/adjoint_common.jl

```diff
@@ -143,8 +146,50 @@ function adjointdiffcache(g,sensealg,discrete,sol,dg,f;quad=false)
   dg_val .= false
   f_cache = DiffEqBase.isinplace(prob) ? deepcopy(u0) : nothing
 
+  if noiseterm
+    if (sensealg.noise isa ReverseDiffNoise ||
+    (sensealg.noise isa Bool && sensealg.noise && DiffEqBase.isinplace(prob)))
+
+      if DiffEqBase.isinplace(prob)
+        tape = ReverseDiff.JacobianTape((y, prob.p, [tspan[2]])) do u,p,t
```

Comment:
can you also make it a function of the idx here?

Reply:
I initially tried something like:
```
indx = 1
tape = ReverseDiff.JacobianTape((y, prob.p, [tspan[2]], [indx])) do u,p,t,i
 		f(u,p,first(t))[i] (and also f(u,p,first(t))[first(i)] )
end
```
to be as close as possible to the Zygote version. But this fails already at creating the tape:
 ```
ArgumentError: invalid index: TrackedReal<1wd>(1, 0.0, E6S, 1, 3Qw) of type ReverseDiff.TrackedReal{Int64,Float64,ReverseDiff.TrackedArray{Int64,Float64,1,Array{Int64,1},Array{Float64,1}}}
Stacktrace:
 [1] to_index(::ReverseDiff.TrackedReal{Int64,Float64,ReverseDiff.TrackedArray{Int64,Float64,1,Array{Int64,1},Array{Float64,1}}}) at ./indices.jl:297
 [2] to_index(::Array{ReverseDiff.TrackedReal{Float64,Float64,Nothing},1}, ::ReverseDiff.TrackedReal{Int64,Float64,ReverseDiff.TrackedArray{Int64,Float64,1,Array{Int64,1},Array{Float64,1}}}) at ./indices.jl:274
 [3] to_indices at ./indices.jl:325 [inlined]
 [4] to_indices at ./indices.jl:322 [inlined]
 [5] getindex at ./abstractarray.jl:980 [inlined]
 [6] (::var"#39#40")(::ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}}, ::ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}}, ::ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}}, ::ReverseDiff.TrackedArray{Int64,Float64,1,Array{Int64,1},Array{Float64,1}}) at ./none:2
 [7] ReverseDiff.JacobianTape(::Function, ::Tuple{Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Int64,1}}, ::ReverseDiff.JacobianConfig{Tuple{ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}},ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}},ReverseDiff.TrackedArray{Float64,Float64,1,Array{Float64,1},Array{Float64,1}},ReverseDiff.TrackedArray{Int64,Float64,1,Array{Int64,1},Array{Float64,1}}},Nothing}) at /Users/frank/.julia/packages/ReverseDiff/uy0uk/src/api/tape.jl:237
 [8] ReverseDiff.JacobianTape(::Function, ::Tuple{Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Int64,1}}) at /Users/frank/.julia/packages/ReverseDiff/uy0uk/src/api/tape.jl:234
 [9] top-level scope at none:0

```
So including the indx directly in the tape didn't work. Alternatively, I could try to make the D tape version from the code snipped in the first comment work.

Reply:
I see, yeah it tries to differentiate through it which isn't good behavior, so yeah it does need separate tapes and the strategy above makes sense.